code/modules/power/supermatter/supermatter.dm

```diff
@@ -678,6 +680,22 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		if(produces_gas)
 			env.merge(removed)
 			air_update_turf()
+	if(takes_damage)
+		for(var/t in RANGE_TURFS(1, loc))
+			if(!isspaceturf(t))
+				continue
+			var/turf/turf_to_check = t
+			if(LAZYLEN(turf_to_check.atmos_adjacent_turfs))
+				var/integrity = get_integrity()
+				if(integrity < 10)
+					damage += clamp((power * 0.0005) * DAMAGE_INCREASE_MULTIPLIER, 0, MAX_SPACE_EXPOSURE_DAMAGE)
+				else if(integrity < 25)
+					damage += clamp((power * 0.0009) * DAMAGE_INCREASE_MULTIPLIER, 0, MAX_SPACE_EXPOSURE_DAMAGE)
+				else if(integrity < 45)
+					damage += clamp((power * 0.005) * DAMAGE_INCREASE_MULTIPLIER, 0, MAX_SPACE_EXPOSURE_DAMAGE)
+				else if(integrity < 75)
+					damage += clamp((power * 0.002) * DAMAGE_INCREASE_MULTIPLIER, 0, MAX_SPACE_EXPOSURE_DAMAGE)
+				break
```

Comment:
Move this whole block up to the same area as 537, so that the max damage the sm can take stays at 2 per half second.